src/metabase/driver/drill.clj

```diff
@@ -0,0 +1,196 @@
+(extend-protocol IPrepareValue
+  nil (prepare-value [_] nil)
+  DateTimeValue (prepare-value [{:keys [value]}] (prepare-value value))
+  Value (prepare-value [{:keys [value]}] (prepare-value value))
+  String (prepare-value [this] (hx/literal this))
+  Boolean (prepare-value [this] (hsql/raw (if this "TRUE" "FALSE")))
+  Date (prepare-value [this] (hsql/call :to_timestamp
+                                        (hx/literal (u/date->iso-8601 this))
+                                        (hx/literal "YYYY-MM-dd''T''HH:mm:ss.SSSZ")))
+  Number (prepare-value [this] this)
+  Object (prepare-value [this] (throw (Exception. (format "Don't know how to prepare value %s %s" (class this) this)))))
+
+(defn date-interval [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s(%d))" (int amount) (name unit)
+                    (count (str amount)))))
+
+(defrecord DrillDriver []
+  clojure.lang.Named
+  (getName [_] "Drill"))
+
+(u/strict-extend DrillDriver
+                 driver/IDriver
+                 (merge (sql/IDriverSQLDefaultsMixin)
+                        {:can-connect? (u/drop-first-arg can-connect?)
+                         :date-interval (u/drop-first-arg date-interval)
+                         :describe-table-fks (constantly #{})
+                         :details-fields (constantly [{:name "drill-connect"
+                                                       :display-name "Drill connect string"
+                                                       :default "drillbit=localhost or zk=localhost:2181/drill/cluster-id"}])
+                         :execute-query execute-query
+                         :features (constantly #{:basic-aggregations
+                                                 :standard-deviation-aggregations
+                                                 ;;:foreign-keys
+                                                 :expressions
+                                                 :expression-aggregations
+                                                 :native-parameters})})
+                 sql/ISQLDriver
+                 (merge (sql/ISQLDriverDefaultsMixin)
+                        {:apply-aggregation bigquery/apply-aggregation
+                         :column->base-type (u/drop-first-arg column->base-type)
+                         :connection-details->spec (u/drop-first-arg connection-details->spec)
+                         :date (u/drop-first-arg date)
+                         :field->identifier (u/drop-first-arg hive-like/field->identifier)
+                         :prepare-value (u/drop-first-arg prepare-value)
+                         :quote-style (constantly :mysql)
+                         :current-datetime-fn (u/drop-first-arg (constantly hive-like/now))
+                         :string-length-fn (u/drop-first-arg hive-like/string-length-fn)
+                         :unix-timestamp->timestamp (u/drop-first-arg hive-like/unix-timestamp->timestamp)}))
```

Comment:
This indentation is a bit weird check out the way the same sorts of things are indented in other drivers. e.g. `driver/IDriver` and the stuff below it should be indented 2 spaces instead of being even with `DrillDriver` at the top

Reply:
Done, I think - I'm finding the indentation a bit challenging. I'm using Emacs but the indentation settings that are applied from your project do not seem to match current practice.

Reply:
We've added `:style/indent` metadata to almost all of the functions where we want things to be indented a certain way, so CIDER should take care of things for you when you reindent the form